### PR TITLE
[core] Minor update on task manager

### PR DESCRIPTION
**Pull request by dentiny**
As usual, perform some minor C++ optimization / cleanup for task manager.

### CODE REVIEW

File: src/ray/core_worker/task_manager.cc [hunk 1]
```diff
@@ -335,45 +337,48 @@ bool TaskManager::ResubmitTask(const TaskID &task_id, std::vector<ObjectID> *tas
...
-      if (spec.ArgByRef(i)) {
-        task_deps->push_back(spec.ArgId(i));
-      } else {
-        const auto &inlined_refs = spec.ArgInlinedRefs(i);
-        for (const auto &inlined_ref : inlined_refs) {
-          task_deps->push_back(ObjectID::FromBinary(inlined_ref.object_id()));
-        }
-      }
-    }
+  if (!resubmit) {
```
Comment (dentiny) [inline src/ray/core_worker/task_manager.cc:340]: Main change here, follows best practice to early return.

File: src/ray/core_worker/task_manager.cc [hunk 2]
```diff
@@ -304,6 +304,8 @@ std::vector<rpc::ObjectReference> TaskManager::AddPendingTask(
 }
 
 bool TaskManager::ResubmitTask(const TaskID &task_id, std::vector<ObjectID> *task_deps) {
+  RAY_CHECK(task_deps->empty());
```
Comment (dentiny) [inline src/ray/core_worker/task_manager.cc:307]: Add an assertion before emplace.

File: src/ray/core_worker/task_manager.h [hunk 3]
```diff
@@ -268,8 +268,9 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
   /// responsible for making sure that these dependencies become available, so
   /// that the resubmitted task can run. This is only populated if the task was
   /// not already pending and was successfully resubmitted.
-  /// \return OK if the task was successfully resubmitted or was
-  /// already pending, Invalid if the task spec is no longer present.
+  /// \return true if the task was successfully resubmitted (task or actor being
```
Comment (dentiny) [inline src/ray/core_worker/task_manager.h:271]: @jjyao I actually have a question on how we retry and retrieve retry result, task manager only *submit* task, but not blocking wait their completion; task recovery manager also doesn't block wait. Is there any concern?

Comment (jjyao) [reply src/ray/core_worker/task_manager.h:271]: Discussed offline. It doesn't block, it's async operation.

**Review by jjyao — APPROVED** [review]

**Pull request merged**